### ISSUE TITLE
UnhandledPromiseRejectionWarning: TypeError: Cannot read property '0' of undefined

### ISSUE THREAD

**Issue by pniedzwiedzinski** [storybookjs/storybook#9908]
**Describe the bug**
When I build storybook I got this error. But when I run storybook with `start-storybook` everything works fine.

**To Reproduce**
1. Go on to https://github.com/zsk-poznan/ui
2. Clone
3. `npm install -D`
4. `npm run build-storybook`

**Expected behavior**
I want my storybook to build

**Screenshots**
```
info => Compiling preview..
0% compiling(node:62840) UnhandledPromiseRejectionWarning: TypeError: Cannot read property '0' of undefined
    at Function.isWebpack4 (/Users/pniedzwiedzinski/code/zsk-ui/node_modules/terser-webpack-plugin/dist/index.js:155:28)
    at /Users/pniedzwiedzinski/code/zsk-ui/node_modules/terser-webpack-plugin/dist/index.js:418:25
    at SyncHook.eval [as call] (eval at create (/Users/pniedzwiedzinski/code/zsk-ui/node_modules/@storybook/core/node_modules/tapable/lib/HookCodeFactory.js:19:10), <anonymous>:89:1)
    at SyncHook.lazyCompileHook (/Users/pniedzwiedzinski/code/zsk-ui/node_modules/@storybook/core/node_modules/tapable/lib/Hook.js:154:20)
    at Compiler.newCompilation (/Users/pniedzwiedzinski/code/zsk-ui/node_modules/@storybook/core/node_modules/webpack/lib/Compiler.js:631:26)
    at /Users/pniedzwiedzinski/code/zsk-ui/node_modules/@storybook/core/node_modules/webpack/lib/Compiler.js:667:29
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/Users/pniedzwiedzinski/code/zsk-ui/node_modules/@storybook/core/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:4:1)
    at AsyncSeriesHook.lazyCompileHook (/Users/pniedzwiedzinski/code/zsk-ui/node_modules/@storybook/core/node_modules/tapable/lib/Hook.js:154:20)
    at Compiler.compile (/Users/pniedzwiedzinski/code/zsk-ui/node_modules/@storybook/core/node_modules/webpack/lib/Compiler.js:662:28)
    at /Users/pniedzwiedzinski/code/zsk-ui/node_modules/@storybook/core/node_modules/webpack/lib/Compiler.js:321:11
    at Compiler.readRecords (/Users/pniedzwiedzinski/code/zsk-ui/node_modules/@storybook/core/node_modules/webpack/lib/Compiler.js:529:11)
    at /Users/pniedzwiedzinski/code/zsk-ui/node_modules/@storybook/core/node_modules/webpack/lib/Compiler.js:318:10
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/Users/pniedzwiedzinski/code/zsk-ui/node_modules/@storybook/core/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:4:1)
    at AsyncSeriesHook.lazyCompileHook (/Users/pniedzwiedzinski/code/zsk-ui/node_modules/@storybook/core/node_modules/tapable/lib/Hook.js:154:20)
    at /Users/pniedzwiedzinski/code/zsk-ui/node_modules/@storybook/core/node_modules/webpack/lib/Compiler.js:315:19
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/Users/pniedzwiedzinski/code/zsk-ui/node_modules/@storybook/core/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:13:1)
(node:62840) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(node:62840) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Code snippets**
If applicable, add code samples to help explain your problem.

**System:**
```
Environment Info:

  System:
    OS: macOS Mojave 10.14.6
    CPU: (4) x64 Intel(R) Core(TM) i5-5350U CPU @ 1.80GHz
  Binaries:
    Node: 13.5.0 - /usr/local/bin/node
    Yarn: 2.0.0-rc.27 - /usr/local/bin/yarn
    npm: 6.13.4 - /usr/local/bin/npm
  Browsers:
    Chrome: 79.0.3945.130
    Safari: 12.1.2
```

**Additional context**

**Comment by pniedzwiedzinski** [comment]
I am using create-react-library and I think that’s the problem here. I tested my configuration on project created with create-react-app and it works.

**Comment by FabzAn** [comment]
Hi, I had the same problem.

The error is emitted by `terser-webpack-plugin` which seemed to be unable to load webpack.
I looked at its dependency tree using `npm ls terser-webpack-plugin`.
It returned `UNMET PEER DEPENDENCY webpack@4.41.6`. Turned out, my installed webpack version was 3.x.x and a simple upgrade fixed the problem.

tl;dr
Try `npm i -D webpack@4.41.6`
